codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java

```diff
@@ -512,4 +512,91 @@ public void testAccountForHeaderOverhead() throws Exception {
             in.release();
         }
     }
+
+    @Test
+    public void unknownPseudoHeader() throws Exception {
+        ByteBuf in = Unpooled.buffer(200);
+        try {
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(":test", "1");
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            // decode the headers, we should get an exception, but
+            // the decoded headers object should contain all of the headers
+            Http2Headers decoded = new DefaultHttp2Headers();
+
+            expectedException.expect(Http2Exception.class);
+            hpackDecoder.decode(1, in, decoded);
+        } finally {
+            in.release();
+        }
+    }
+
+    @Test
+    public void requestPseudoHeaderInResponse() throws Exception {
+        ByteBuf in = Unpooled.buffer(200);
+        try {
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(":status", "200");
+            toEncode.add(":method", "GET");
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            // decode the headers, we should get an exception, but
+            // the decoded headers object should contain all of the headers
+            Http2Headers decoded = new DefaultHttp2Headers();
+
+            expectedException.expect(Http2Exception.class);
+            hpackDecoder.decode(1, in, decoded);
+        } finally {
+            in.release();
```

Comment:
should we verify that `decoded` contains `:status=200`?

Reply:
I think it's only necessary if we expect consumers to use the `Http2Headers` after a failed decoding attempt. It's currently not the case in `DefaultHttp2HeadersDecoder`

Reply:
sgtm